nilearn/maskers/multi_nifti_masker.py

```diff
@@ -246,23 +253,31 @@ def fit(self, imgs=None, y=None):
                               ' requested (imgs != None) while a mask has'
                               ' been provided at masker creation. Given mask'
                               ' will be used.' % self.__class__.__name__)
+
             self.mask_img_ = _utils.check_niimg_3d(self.mask_img)
 
         # If resampling is requested, resample the mask as well.
         # Resampling: allows the user to change the affine, the shape or both.
         if self.verbose > 0:
             print("[%s.transform] Resampling mask" % self.__class__.__name__)
+
         self.mask_img_ = self._cache(image.resample_img)(
             self.mask_img_,
             target_affine=self.target_affine,
             target_shape=self.target_shape,
             interpolation='nearest', copy=False)
+
         if self.target_affine is not None:
             self.affine_ = self.target_affine
         else:
             self.affine_ = self.mask_img_.affine
+
         # Load data in memory
-        get_data(self.mask_img_)
+        data = get_data(self.mask_img_)
+
+        # Infer the number of elements (voxels) in the mask
+        self.n_elements_ = int(data.sum())
```

Comment:
Silly question maybe, but is the mask always binary ? We need to check whether the mask is forced to be binary.

Reply:
I assumed it was, but it looks like that is only checked at `transform` via `_load_mask_img`. Should I add a check in `fit` as well?

Reply:
I swapped out `get_data` with `_load_mask_img` in `fit`, which will check that the mask is binary.

Reply:
It looks like the tests `test_nifti_masker.py::test_compute_brain_mask` and `test_multi_nifti_masker.py::test_compute_multi_gray_matter_mask` have multiple checks that explicitly allow empty masks. @NicolasGensollen it looks like you added these tests in #2904. Were they just smoke tests, or do we want to allow empty masks at `fit`?

Reply:
My gut feeling is that we show allow empty masks.

Reply:
I think there's a typo in your comment. Are you saying "should allow" or "shouldn't allow"?

Reply:
Sorry: we *should* allow.

Reply:
Ah, okay. What would the expected behavior be in that case? Should (1) fit pass and transform fail, (2) fit and transform pass, but transform returns an array of NaNs, or (3) fit and transform pass, but transform returns an empty array?

Reply:
The last one makes more sense to me.

Reply:
I think most of that will need to be tackled in a separate PR, but for this one I can ensure that an exception is not raised when the mask is empty.